returnn/torch/engine.py

```diff
@@ -51,7 +51,15 @@ def __init__(self, config: Config):
         self._save_model_epoch_interval = 1
         self._updater = None  # type: Optional[Updater]
 
-        self._device = "cuda" if torch.cuda.is_available() and torch.cuda.device_count() > 0 else "cpu"
+        device = config.typed_value("device")
+        if (device == "gpu"):
+            print("Use of deprecated keyword 'gpu' to set device. Using 'cuda' instead.")
+            device = "cuda"
+
+        self._device = device
+        assert (self._device == "cuda" and torch.cuda.is_available() and torch.cuda.device_count() > 0) or config.typed_value(
```

Comment:
I would recommend to do:
```
        if self._device == "cuda":
            assert (torch.cuda.is_available() and torch.cuda.device_count() > 0) , ...
```
This way we would allow for more devices than just `cuda` and `cpu`, PyTorch has quite the long list:
` cpu, cuda, ipu, xpu, mkldnn, opengl, opencl, ideep, hip, ve, fpga, ort, xla, lazy, vulkan, mps, meta, hpu, privateuseone`